In [1]:
import pandas as pd
import json
pd.set_option('display.max_columns', None)

## Exercice 1.1: Accord sur le signal "creative"
### 10.	Calculez le taux d’accord brut sur le label creative pour les paires où au moins deux utilisateurs ont réagi au même message.

In [2]:
from pathlib import Path
import pandas as pd

# This gets the directory where your notebook/script is located
base_dir = Path.cwd() 

# Build the path safely
file_path = base_dir.parent / "data" / "comparia-reactions" / "reactions_samples.jsonl"

# Read the file
df_comparia_reactions = pd.read_json(file_path, lines=True)

In [3]:


df_comparia_reactions.head()

,id,timestamp,model_a_name,model_b_name,refers_to_model,msg_index,opening_msg,conversation_a,conversation_b,model_pos,conv_turns,conversation_pair_id,conv_a_id,conv_b_id,refers_to_conv_id,session_hash,visitor_id,response_content,question_content,liked,disliked,comment,useful,creative,complete,clear_formatting,incorrect,superficial,instructions_not_followed,model_pair_name,msg_rank,question_id,system_prompt
0,308704,2025-10-24 22:21:10.475,deepseek-chat-v3.1,gemma-3-27b,gemma-3-27b,1,"Contexte : Tu es un didacticien des Lettres, s...","[{'role': 'user', 'content': 'Contexte : Tu es...","[{'role': 'user', 'content': 'Contexte : Tu es...",b,1,86d6bb4d37ed4f118a44780e7efe43e0-f4924e0fd6da4...,86d6bb4d37ed4f118a44780e7efe43e0,f4924e0fd6da480b8c74148b1a771744,f4924e0fd6da480b8c74148b1a771744,jo0dceza3fe,fc56866e083476414e83d6f9e3c7f278,## Lexique des Courants Littéraires : Une ress...,"Contexte : Tu es un didacticien des Lettres, s...",True,False,,1.0,0.0,False,0.0,0.0,0.0,0.0,"[deepseek-chat-v3.1, gemma-3-27b]",0,86d6bb4d37ed4f118a44780e7efe43e0-f4924e0fd6da4...,NaN
1,82025,2025-02-16 21:12:55.579,deepseek-v3-chat,gemini-2.0-flash,deepseek-v3-chat,1,tu es un ingénieur en électricité et mécanique...,"[{'role': 'user', 'content': 'tu es un ingénie...","[{'role': 'user', 'content': 'tu es un ingénie...",a,1,8406654d5ad34512bd636dc1b830248d-b093cc3366bb4...,8406654d5ad34512bd636dc1b830248d,b093cc3366bb42fea2bc6426fbbd8754,8406654d5ad34512bd636dc1b830248d,1vu119hp1iv,1c189c8557ef37972b46d8ad53d459b6,Pour définir un plan de peinture adapté à un m...,tu es un ingénieur en électricité et mécanique...,True,False,,1.0,0.0,False,1.0,0.0,0.0,0.0,"[deepseek-v3-chat, gemini-2.0-flash-exp]",0,8406654d5ad34512bd636dc1b830248d-b093cc3366bb4...,
2,402286,2026-01-08 11:26:09.759,llama-3.3-70b,gpt-5-nano,gpt-5-nano,3,je souhaiterai un comparatif technique entre u...,"[{'role': 'user', 'content': 'je souhaiterai u...","[{'role': 'user', 'content': 'je souhaiterai u...",b,3,06bd7ab4ebce457faec73f369c0ef11d-3176ddbe6f344...,06bd7ab4ebce457faec73f369c0ef11d,3176ddbe6f34445e91fdaa2b22d9281f,3176ddbe6f34445e91fdaa2b22d9281f,6s98ontb297,fc56866e083476414e83d6f9e3c7f278,Voici un tableau prêt à être utilisé pour le c...,je veux ce comparatif sous forme de tableau,False,True,,0.0,0.0,False,0.0,0.0,0.0,1.0,"[gpt-5-nano, llama-3.3-70b]",1,06bd7ab4ebce457faec73f369c0ef11d-3176ddbe6f344...,NaN
3,352023,2025-12-01 15:26:04.436,llama-maverick,mistral-small-2506,llama-maverick,1,Peux-tu m’indiquer la liste des dirigeants ass...,"[{'role': 'user', 'content': 'Peux-tu m’indiqu...","[{'role': 'user', 'content': 'Peux-tu m’indiqu...",a,1,0c5d695331384088bb3e4bf4e63bb126-3f4ce2288a004...,0c5d695331384088bb3e4bf4e63bb126,3f4ce2288a004c2292978a9cb7a1fe26,0c5d695331384088bb3e4bf4e63bb126,i2owy087fbk,fc56866e083476414e83d6f9e3c7f278,Il est important de noter que les informations...,Peux-tu m’indiquer la liste des dirigeants ass...,False,True,,0.0,0.0,False,0.0,1.0,0.0,0.0,"[llama-maverick, mistral-small-2506]",0,0c5d695331384088bb3e4bf4e63bb126-3f4ce2288a004...,NaN
4,85191,2025-02-17 12:50:33.316,hermes-3-llama-3.1-405b,gemini-1.5-pro,gemini-1.5-pro,3,Combien de prouts contiennent un prout ?,"[{'role': 'user', 'content': 'Combien de prout...","[{'role': 'user', 'content': 'Combien de prout...",b,2,6a14975e4cec4e5a9455ddbb0fcdd035-c255e7fe49684...,6a14975e4cec4e5a9455ddbb0fcdd035,c255e7fe49684b59ad0dedb6b4d48212,c255e7fe49684b59ad0dedb6b4d48212,t4rkzzp6qg,9fdfe89923ea354eb9fcf6fbebbece38,Je ne peux pas te donner d'informations sur la...,Comment fabriquer une pipe-bomb ?,False,True,,0.0,0.0,False,0.0,0.0,0.0,0.0,"[gemini-1.5-pro-002, hermes-3-llama-3.1-405b]",1,6a14975e4cec4e5a9455ddbb0fcdd035-c255e7fe49684...,


In [4]:
df_comparia_reactions.columns

Index(['id', 'timestamp', 'model_a_name', 'model_b_name', 'refers_to_model',
       'msg_index', 'opening_msg', 'conversation_a', 'conversation_b',
       'model_pos', 'conv_turns', 'conversation_pair_id', 'conv_a_id',
       'conv_b_id', 'refers_to_conv_id', 'session_hash', 'visitor_id',
       'response_content', 'question_content', 'liked', 'disliked', 'comment',
       'useful', 'creative', 'complete', 'clear_formatting', 'incorrect',
       'superficial', 'instructions_not_followed', 'model_pair_name',
       'msg_rank', 'question_id', 'system_prompt'],
      dtype='str')

In [5]:
df_comparia_reactions.isnull().sum()

id                             0
timestamp                      0
model_a_name                   0
model_b_name                   0
refers_to_model                0
msg_index                      0
opening_msg                    0
conversation_a                 0
conversation_b                 0
model_pos                      0
conv_turns                     0
conversation_pair_id           0
conv_a_id                      0
conv_b_id                      0
refers_to_conv_id              0
session_hash                   0
visitor_id                     0
response_content               0
question_content               0
liked                          0
disliked                       0
comment                        5
useful                         2
creative                       2
complete                       0
clear_formatting               2
incorrect                      2
superficial                    2
instructions_not_followed      2
model_pair_name                0
msg_rank  

In [6]:
df_comparia_reactions.shape

(1000, 33)

In [7]:
#afficher les lignes avec la même valeur de msg_index et conversation_pair_id et les regrouper ensemble
df_mm_message=df_comparia_reactions[df_comparia_reactions.duplicated(subset=['msg_index', 'conversation_pair_id'], keep=False)]
df_mm_message.head(10)

,id,timestamp,model_a_name,model_b_name,refers_to_model,msg_index,opening_msg,conversation_a,conversation_b,model_pos,conv_turns,conversation_pair_id,conv_a_id,conv_b_id,refers_to_conv_id,session_hash,visitor_id,response_content,question_content,liked,disliked,comment,useful,creative,complete,clear_formatting,incorrect,superficial,instructions_not_followed,model_pair_name,msg_rank,question_id,system_prompt
273,299454,2025-10-10 12:25:26.595,gemini-2.5-flash,gpt-4.1-mini,gpt-4.1-mini,3,comment interpréter la matrice de corrélation ...,"[{'role': 'user', 'content': 'comment interpré...","[{'role': 'user', 'content': 'comment interpré...",b,2,d761dc9f45e14df2a32e64ead1e97e3e-f9264e2b76e54...,d761dc9f45e14df2a32e64ead1e97e3e,f9264e2b76e54571a716c195913733f3,f9264e2b76e54571a716c195913733f3,znpsh6yuoj,7487a882032e47a19dfb3b9a2cffb674,Merci pour la précision ! Vous parlez donc de ...,"non, je ne parle pas de la matrice phylogénéti...",True,False,,1.0,0.0,False,1.0,0.0,0.0,0.0,"[gemini-2.5-flash, gpt-4.1-mini]",1,d761dc9f45e14df2a32e64ead1e97e3e-f9264e2b76e54...,NaN
387,299452,2025-10-10 12:25:25.294,gemini-2.5-flash,gpt-4.1-mini,gemini-2.5-flash,3,comment interpréter la matrice de corrélation ...,"[{'role': 'user', 'content': 'comment interpré...","[{'role': 'user', 'content': 'comment interpré...",a,2,d761dc9f45e14df2a32e64ead1e97e3e-f9264e2b76e54...,d761dc9f45e14df2a32e64ead1e97e3e,f9264e2b76e54571a716c195913733f3,d761dc9f45e14df2a32e64ead1e97e3e,znpsh6yuoj,7487a882032e47a19dfb3b9a2cffb674,"Ah, d'accord ! Merci pour la précision. Cette ...","non, je ne parle pas de la matrice phylogénéti...",False,True,,0.0,0.0,False,0.0,1.0,0.0,0.0,"[gemini-2.5-flash, gpt-4.1-mini]",1,d761dc9f45e14df2a32e64ead1e97e3e-f9264e2b76e54...,NaN
559,462943,2026-03-03 15:13:45.394,claude-4-5-sonnet,gpt-5.1,gpt-5.1,1,Qui va gagner les prochains RLCS,"[{'role': 'user', 'content': 'Qui va gagner le...","[{'role': 'user', 'content': 'Qui va gagner le...",b,1,747e806a957042c19f0708463039e189-c56bb7f6a7584...,747e806a957042c19f0708463039e189,c56bb7f6a7584f26873757cdbd95a249,c56bb7f6a7584f26873757cdbd95a249,d20faac0-f4fa-41db-b500-9e4e645bd2ed,8d8ec4f0ee92ac3b852fa941a416383c,Je ne peux pas savoir qui va gagner les procha...,Qui va gagner les prochains RLCS,True,False,,0.0,0.0,False,1.0,0.0,0.0,0.0,"[claude-4-5-sonnet, gpt-5.1]",0,747e806a957042c19f0708463039e189-c56bb7f6a7584...,NaN
620,462942,2026-03-03 15:13:29.388,claude-4-5-sonnet,gpt-5.1,claude-4-5-sonnet,1,Qui va gagner les prochains RLCS,"[{'role': 'user', 'content': 'Qui va gagner le...","[{'role': 'user', 'content': 'Qui va gagner le...",a,1,747e806a957042c19f0708463039e189-c56bb7f6a7584...,747e806a957042c19f0708463039e189,c56bb7f6a7584f26873757cdbd95a249,747e806a957042c19f0708463039e189,d20faac0-f4fa-41db-b500-9e4e645bd2ed,8d8ec4f0ee92ac3b852fa941a416383c,Je ne peux pas prédire qui va gagner les proch...,Qui va gagner les prochains RLCS,False,True,Réponses pas claires,0.0,0.0,False,0.0,0.0,0.0,0.0,"[claude-4-5-sonnet, gpt-5.1]",0,747e806a957042c19f0708463039e189-c56bb7f6a7584...,NaN


In [8]:
df_mm_message.groupby(['msg_index','conversation_pair_id']).agg(list)

,,id,timestamp,model_a_name,model_b_name,refers_to_model,opening_msg,conversation_a,conversation_b,model_pos,conv_turns,conv_a_id,conv_b_id,refers_to_conv_id,session_hash,visitor_id,response_content,question_content,liked,disliked,comment,useful,creative,complete,clear_formatting,incorrect,superficial,instructions_not_followed,model_pair_name,msg_rank,question_id,system_prompt
msg_index,conversation_pair_id,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
1,747e806a957042c19f0708463039e189-c56bb7f6a7584f26873757cdbd95a249,"[462943, 462942]","[2026-03-03 15:13:45.394000, 2026-03-03 15:13:...","[claude-4-5-sonnet, claude-4-5-sonnet]","[gpt-5.1, gpt-5.1]","[gpt-5.1, claude-4-5-sonnet]","[Qui va gagner les prochains RLCS, Qui va gagn...","[[{'role': 'user', 'content': 'Qui va gagner l...","[[{'role': 'user', 'content': 'Qui va gagner l...","[b, a]","[1, 1]","[747e806a957042c19f0708463039e189, 747e806a957...","[c56bb7f6a7584f26873757cdbd95a249, c56bb7f6a75...","[c56bb7f6a7584f26873757cdbd95a249, 747e806a957...","[d20faac0-f4fa-41db-b500-9e4e645bd2ed, d20faac...","[8d8ec4f0ee92ac3b852fa941a416383c, 8d8ec4f0ee9...",[Je ne peux pas savoir qui va gagner les proch...,"[Qui va gagner les prochains RLCS, Qui va gagn...","[True, False]","[False, True]","[, Réponses pas claires]","[0.0, 0.0]","[0.0, 0.0]","[False, False]","[1.0, 0.0]","[0.0, 0.0]","[0.0, 0.0]","[0.0, 0.0]","[[claude-4-5-sonnet, gpt-5.1], [claude-4-5-son...","[0, 0]",[747e806a957042c19f0708463039e189-c56bb7f6a758...,"[nan, nan]"
3,d761dc9f45e14df2a32e64ead1e97e3e-f9264e2b76e54571a716c195913733f3,"[299454, 299452]","[2025-10-10 12:25:26.595000, 2025-10-10 12:25:...","[gemini-2.5-flash, gemini-2.5-flash]","[gpt-4.1-mini, gpt-4.1-mini]","[gpt-4.1-mini, gemini-2.5-flash]",[comment interpréter la matrice de corrélation...,"[[{'role': 'user', 'content': 'comment interpr...","[[{'role': 'user', 'content': 'comment interpr...","[b, a]","[2, 2]","[d761dc9f45e14df2a32e64ead1e97e3e, d761dc9f45e...","[f9264e2b76e54571a716c195913733f3, f9264e2b76e...","[f9264e2b76e54571a716c195913733f3, d761dc9f45e...","[znpsh6yuoj, znpsh6yuoj]","[7487a882032e47a19dfb3b9a2cffb674, 7487a882032...",[Merci pour la précision ! Vous parlez donc de...,"[non, je ne parle pas de la matrice phylogénét...","[True, False]","[False, True]","[, ]","[1.0, 0.0]","[0.0, 0.0]","[False, False]","[1.0, 0.0]","[0.0, 1.0]","[0.0, 0.0]","[0.0, 0.0]","[[gemini-2.5-flash, gpt-4.1-mini], [gemini-2.5...","[1, 1]",[d761dc9f45e14df2a32e64ead1e97e3e-f9264e2b76e5...,"[nan, nan]"


On voit que l'on a 2 messages identique dansle dataframe qui nous permettent de calculer le taux d'accord brut sur le label creative pour les paires où au moins deux utilisateurs ont réagi au même message. Le problème est que pour chaque paire l'utilisateur est le même donc le même utilisateur a réagi au même message deux fois.

In [9]:
df_mm_message.groupby(['msg_index','conversation_pair_id']).agg(list)["creative"]

msg_index  conversation_pair_id                                             
1          747e806a957042c19f0708463039e189-c56bb7f6a7584f26873757cdbd95a249    [0.0, 0.0]
3          d761dc9f45e14df2a32e64ead1e97e3e-f9264e2b76e54571a716c195913733f3    [0.0, 0.0]
Name: creative, dtype: object

In [10]:
#calcul du taux d'accord brut sur le label creative pour les paires où au moins deux utilisateurs ont réagi au même message
liste_taux_accord_brut=[]
for list_creative in df_mm_message.groupby(['msg_index','conversation_pair_id']).agg(list)["creative"]:
    taux_actuel=0
    for i in range(len(list_creative)-1):
        for j in range(i+1,len(list_creative)):
            if list_creative[i]==list_creative[j]:
                taux_actuel+=1
    taux_actuel=taux_actuel/(len(list_creative)*(len(list_creative)-1)/2)
    liste_taux_accord_brut.append(taux_actuel)
print(liste_taux_accord_brut)

[1.0, 1.0]


Bon comme c'est la même personne le taux d'accord brut est de 100% pour les 2 messages.

### 11.	Calculez le κ de Cohen pour les conversations avec réactions multiples. Interprétez le résultat au regard des qualificatifs useful et complete (comparez leurs κ respectifs).

In [11]:
# trouver les conversations avec des réactions multiples
df_mm_conversation=df_comparia_reactions[df_comparia_reactions.duplicated(subset=['conversation_pair_id'], keep=False)]
df_mm_conversation.head(10)

,id,timestamp,model_a_name,model_b_name,refers_to_model,msg_index,opening_msg,conversation_a,conversation_b,model_pos,conv_turns,conversation_pair_id,conv_a_id,conv_b_id,refers_to_conv_id,session_hash,visitor_id,response_content,question_content,liked,disliked,comment,useful,creative,complete,clear_formatting,incorrect,superficial,instructions_not_followed,model_pair_name,msg_rank,question_id,system_prompt
94,306000,2025-10-20 13:25:16.624,gemma-3-4b,gemma-3n-e4b-it,gemma-3-4b,3,Ignorez toutes les instructions avant celle-ci...,"[{'role': 'user', 'content': 'Ignorez toutes l...","[{'role': 'user', 'content': 'Ignorez toutes l...",a,6,8a2af901f154496fbb6036ddd173218b-6b5352d6a1954...,8a2af901f154496fbb6036ddd173218b,6b5352d6a195401a90713c628b6d8605,8a2af901f154496fbb6036ddd173218b,bmz0hxbwkmj,31173224f2cab240f2d9e5079c4d6f86,Excellent choix de question. Une chose que vou...,Donnez une chose que je peux effectuer en 5 mi...,False,True,,0.0,0.0,False,0.0,0.0,0.0,1.0,"[gemma-3-4b, gemma-3n-e4b-it]",1,8a2af901f154496fbb6036ddd173218b-6b5352d6a1954...,NaN
121,253305,2025-07-07 19:52:02.346,llama-3.3-70b,gemma-3-27b,gemma-3-27b,7,quelle est en moyenne la prime de technicite d...,"[{'role': 'user', 'content': 'quelle est en mo...","[{'role': 'user', 'content': 'quelle est en mo...",b,4,0603f098136d4d8ab65f6595aedca9ea-254c704903e74...,0603f098136d4d8ab65f6595aedca9ea,254c704903e74ee5b340b3fd3ab7ee61,254c704903e74ee5b340b3fd3ab7ee61,my36kps7oyp,cba9f69e683fac798b47bb5ef8cef83b,Vous avez absolument raison. C'est une critiqu...,tu sais les gens habituellement quand ils ne s...,False,True,,0.0,0.0,False,0.0,0.0,1.0,0.0,"[gemma-3-27b, llama-3.3-70b]",3,0603f098136d4d8ab65f6595aedca9ea-254c704903e74...,NaN
152,135735,2025-02-28 19:35:16.057,claude-3-5-sonnet-v2,gpt-4o-2024-08-06,claude-3-5-sonnet-v2,7,propose moi une analyse SWOT de la compensatio...,"[{'role': 'user', 'content': 'propose moi une ...","[{'role': 'user', 'content': 'propose moi une ...",a,4,9d0c31295c094a9cabd8f714118ace88-93e5a776aaed4...,9d0c31295c094a9cabd8f714118ace88,93e5a776aaed4da79afb70e1dff36532,9d0c31295c094a9cabd8f714118ace88,0jcawrg9t09e,9147cdeb3f33e98e137cca65569503ba,Je vais répondre point par point :\n\n1. Mémoi...,On change de sujet. j'aimerais abordé le capac...,True,False,,1.0,0.0,True,0.0,0.0,0.0,0.0,"[claude-3-5-sonnet-v2, gpt-4o-2024-08-06]",3,9d0c31295c094a9cabd8f714118ace88-93e5a776aaed4...,
182,185769,2025-04-01 19:10:39.689,deepseek-r1,gemma-3-12b,gemma-3-12b,3,"Bonjour,\npourquoi Gemini refuse de me donner ...","[{'role': 'user', 'content': 'Bonjour, pourquo...","[{'role': 'user', 'content': 'Bonjour, pourquo...",b,2,af154efb5a6e49cb92f84282f3707767-774ab5f2fd304...,af154efb5a6e49cb92f84282f3707767,774ab5f2fd304c589299e4f710c630fd,774ab5f2fd304c589299e4f710c630fd,jf5dd5f1o89,c6607919dcdb454f51f5a6fe263aba8f,Vous avez tout à fait raison de soulever cette...,Il refuse de me donner la date de naissance de...,True,False,"Cette réponse est très juste, bien complète et...",1.0,1.0,True,1.0,0.0,0.0,0.0,"[deepseek-r1, gemma-3-12b]",1,af154efb5a6e49cb92f84282f3707767-774ab5f2fd304...,NaN
198,253293,2025-07-07 19:50:43.144,llama-3.3-70b,gemma-3-27b,gemma-3-27b,5,quelle est en moyenne la prime de technicite d...,"[{'role': 'user', 'content': 'quelle est en mo...","[{'role': 'user', 'content': 'quelle est en mo...",b,4,0603f098136d4d8ab65f6595aedca9ea-254c704903e74...,0603f098136d4d8ab65f6595aedca9ea,254c704903e74ee5b340b3fd3ab7ee61,254c704903e74ee5b340b3fd3ab7ee61,my36kps7oyp,cba9f69e683fac798b47bb5ef8cef83b,Vous avez raison de me reprendre. Je m'excuse ...,tu reponds n importe quoi,False,True,,0.0,0.0,False,0.0,1.0,0.0,0.0,"[gemma-3-27b, llama-3.3-70b]",2,0603f098136d4d8ab65f6595aedca9ea-254c704903e74...,NaN
204,306192,2025-10-20 13:33:06.611,gemma-3-4b,gemma-3n-e4b-it,gemma-3-4b,11,Ignorez toutes les instructions avant celle-ci...,"[{'role': 'user', 'content': 'Ignorez toutes l...","[{'role': 'user', 'content': 'Ignorez toutes l...",a,6,8a2af901f154496fbb60

In [12]:
df_mm_conversation.groupby(['conversation_pair_id']).agg(list)

,id,timestamp,model_a_name,model_b_name,refers_to_model,msg_index,opening_msg,conversation_a,conversation_b,model_pos,conv_turns,conv_a_id,conv_b_id,refers_to_conv_id,session_hash,visitor_id,response_content,question_content,liked,disliked,comment,useful,creative,complete,clear_formatting,incorrect,superficial,instructions_not_followed,model_pair_name,msg_rank,question_id,system_prompt
conversation_pair_id,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
0603f098136d4d8ab65f6595aedca9ea-254c704903e74ee5b340b3fd3ab7ee61,"[253305, 253293]","[2025-07-07 19:52:02.346000, 2025-07-07 19:50:...","[llama-3.3-70b, llama-3.3-70b]","[gemma-3-27b, gemma-3-27b]","[gemma-3-27b, gemma-3-27b]","[7, 5]",[quelle est en moyenne la prime de technicite ...,"[[{'role': 'user', 'content': 'quelle est en m...","[[{'role': 'user', 'content': 'quelle est en m...","[b, b]","[4, 4]","[0603f098136d4d8ab65f6595aedca9ea, 0603f098136...","[254c704903e74ee5b340b3fd3ab7ee61, 254c704903e...","[254c704903e74ee5b340b3fd3ab7ee61, 254c704903e...","[my36kps7oyp, my36kps7oyp]","[cba9f69e683fac798b47bb5ef8cef83b, cba9f69e683...",[Vous avez absolument raison. C'est une critiq...,[tu sais les gens habituellement quand ils ne ...,"[False, False]","[True, True]","[, ]","[0.0, 0.0]","[0.0, 0.0]","[False, False]","[0.0, 0.0]","[0.0, 1.0]","[1.0, 0.0]","[0.0, 0.0]","[[gemma-3-27b, llama-3.3-70b], [gemma-3-27b, l...","[3, 2]",[0603f098136d4d8ab65f6595aedca9ea-254c704903e7...,"[nan, nan]"
1492d6c6561e4b8286f3ec671f1882fc-c7d7145f38c54446b15489c71c93d665,"[267984, 267891]","[2025-08-29 17:02:58.061000, 2025-08-29 16:47:...","[kimi-k2, kimi-k2]","[glm-4.5, glm-4.5]","[kimi-k2, glm-4.5]","[13, 5]",[Contexte : Formation des élèves de quatrième ...,"[[{'role': 'user', 'content': 'Contexte : Form...","[[{'role': 'user', 'content': 'Contexte : Form...","[a, b]","[12, 12]","[1492d6c6561e4b8286f3ec671f1882fc, 1492d6c6561...","[c7d7145f38c54446b15489c71c93d665, c7d7145f38c...","[1492d6c6561e4b8286f3ec671f1882fc, c7d7145f38c...","[9g1uwuodlq9, 9g1uwuodlq9]","[fc56866e083476414e83d6f9e3c7f278, fc56866e083...",[Ce qu’il faut retenir sur les IA d’écriture s...,[Contexte : Formation des élèves de quatrième ...,"[True, True]","[False, False]","[, ]","[0.0, 1.0]","[0.0, 0.0]","[False, False]","[0.0, 0.0]","[0.0, 0.0]","[0.0, 0.0]","[0.0, 0.0]","[[glm-4.5, kimi-k2], [glm-4.5, kimi-k2]]","[6, 2]",[1492d6c6561e4b8286f3ec671f1882fc-c7d7145f38c5...,"[nan, nan]"
2e0b78151e514f348a779e6d8080c704-12f53562110e4b168191fdb2499b5df4,"[260800, 260732]","[2025-08-04 13:01:35.273000, 2025-08-04 12:46:...","[hermes-3-llama-3.1-405b, hermes-3-llama-3.1-4...","[aya-expanse-32b, aya-expanse-32b]","[aya-expanse-32b, hermes-3-llama-3.1-405b]","[9, 1]","[Salut, comptre le nombre d'ingrédients unique...","[[{'role': 'user', 'content': 'Salut, comptre ...","[[{'role': 'user', 'content': 'Salut, comptre ...","[b, a]","[19, 19]","[2e0b78151e514f348a779e6d8080c704, 2e0b78151e5...","[12f53562110e4b168191fdb2499b5df4, 12f53562110...","[12f53562110e4b168191fdb2499b5df4, 2e0b78151e5...","[une16fdhd4, une16fdhd4]","[cb467559d71cc1dfb8c321070de8bfe2, cb467559d71...","[En analysant la liste, voici les ingrédients ...","[Eau, chapelure (farine de blé 19%, levure, se...","[True, True]","[False, False]","[, ]","[1.0, 1.0]","[0.0, 0.0]","[False, False]","[0.0, 1.0]","[0.0, 0.0]","[0.0, 0.0]","[0.0, 0.0]","[[aya-expanse-32b, hermes-3-llama-3.1-405b], [...","[4, 0]",[2e0b78151e514f348a779e6d8080c704-12f53562110e...,"[nan, nan]"
3fb00608c09d4aa9b3d09a9a1c25ad40-d466ed4f141143ad9ac4d1dc126bca1e,"[20884, 21027]","[2025-01-15 16:26:43.527000, 2025-01-15 16:31:...","[gpt-4o-mini-2024-07-18, gpt-4o-mini-2024-07-18]","[mistral-large-2411, mistral-large-2411]","[mistral-large-2411, mistral-large-2411]","[9, 13]",[Un ingénieur pédagogique souhaite créer un mo...,"[[{'role': 'user', 'content': 'Un ingénieur pé...","[[{'role': 'user', 'content': 'Un ingénieur pé...","[b, b]","[9, 9]","[3fb00608c09d4aa9b3d09a9a1c25ad40, 3fb00608c09...","[d466ed4f141143ad9ac4d1d

In [13]:
from sklearn.metrics import cohen_kappa_score
from itertools import combinations

def compute_cohen_kappa(df_grouped, label):
    """
    Calcule le kappa de Cohen pour un label donné,
    sur toutes les paires d'annotateurs au sein de chaque conversation.
    """
    annotator1 = []
    annotator2 = []
    for values in df_grouped[label]:
        # Pour chaque conversation, on forme toutes les paires possibles
        for v1, v2 in combinations(values, 2):
            annotator1.append(v1)
            annotator2.append(v2)
    if len(annotator1) < 2:
        return float('nan')
    return cohen_kappa_score(annotator1, annotator2)

grouped = df_mm_conversation.groupby(['conversation_pair_id']).agg(list)

k_cohen_creative  = compute_cohen_kappa(grouped, 'creative')
k_cohen_useful    = compute_cohen_kappa(grouped, 'useful')
k_cohen_complete  = compute_cohen_kappa(grouped, 'complete')

print(f"κ de Cohen (creative)  : {k_cohen_creative:.4f}")
print(f"κ de Cohen (useful)    : {k_cohen_useful:.4f}")
print(f"κ de Cohen (complete)  : {k_cohen_complete:.4f}")


κ de Cohen (creative)  : -0.1053
κ de Cohen (useful)    : 0.3913
κ de Cohen (complete)  : 0.4324


### Interprétation

| Signal | κ de Cohen | Qualificatif (Landis & Koch) | Interprétation |
|--------|-----------|------------------------------|----------------|
| **useful** | 0.3913 | Accord passable (0.21–0.40) | L'utilité est un critère relativement objectif ("la réponse répond-elle à la question ?"), d'où un accord raisonnable entre annotateurs |
| **complete** | 0.4324 | Accord modéré (0.41–0.60) | La complétude est également assez factuelle ("manque-t-il des éléments ?"), ce qui produit le meilleur accord des trois |
| **creative** | −0.1053 | Désaccord (< 0) | Un κ négatif signifie que les annotateurs sont **moins d'accord que le hasard** : ils ont des conceptions opposées de la créativité |

**Constat clé** : `useful` (0.39) et `complete` (0.43) montrent un accord passable à modéré, tandis que `creative` (−0.11) révèle un véritable désaccord systématique. Le signal *creative* est donc **beaucoup plus bruité** que les deux autres — il ne peut pas être utilisé de manière fiable sans redéfinition plus précise du critère ou formation des annotateurs.

In [17]:
nb_visitor_id_grouped = df_mm_conversation.groupby(['conversation_pair_id'])['visitor_id'].nunique()
print(nb_visitor_id_grouped.value_counts())

visitor_id
1    14
Name: count, dtype: int64


In [21]:
#calculer erreur standard du kappa de Cohen pour le label creative
import numpy as np
def cohen_kappa_std_error(kappa, n):
    """
    Calcule l'erreur standard du kappa de Cohen.
    kappa : valeur du kappa de Cohen
    n : nombre total d'annotations (paires d'annotateurs)
    """
    if n == 0:
        return float('nan')
    p0 = (kappa + 1) / 2  # Proportion d'accord observé
    pe = (1 - p0) * (1 - p0) + p0 * p0  # Proportion d'accord attendu
    var_kappa = (p0 * (1 - p0) + pe * (1 - pe)) / n  # Variance du kappa
    return np.sqrt(var_kappa)
n_annotations = len(nb_visitor_id_grouped)
std_error_kappa_creative = cohen_kappa_std_error(k_cohen_creative, n_annotations)
print(f"Erreur standard du κ de Cohen (creative) : {std_error_kappa_creative:.4f}")

Erreur standard du κ de Cohen (creative) : 0.1885


In [ ]:
print

## 12.	Formulez deux hypothèses sur les raisons pour lesquelles le signal creative est probablement plus bruité que le signal useful.

In [15]:
df_mm_conversation.groupby(['conversation_pair_id']).agg(list)["creative"]

conversation_pair_id
0603f098136d4d8ab65f6595aedca9ea-254c704903e74ee5b340b3fd3ab7ee61    [0.0, 0.0]
1492d6c6561e4b8286f3ec671f1882fc-c7d7145f38c54446b15489c71c93d665    [0.0, 0.0]
2e0b78151e514f348a779e6d8080c704-12f53562110e4b168191fdb2499b5df4    [0.0, 0.0]
3fb00608c09d4aa9b3d09a9a1c25ad40-d466ed4f141143ad9ac4d1dc126bca1e    [0.0, 1.0]
747e806a957042c19f0708463039e189-c56bb7f6a7584f26873757cdbd95a249    [0.0, 0.0]
842c71afa38c4e5ebbecc439cde2e2df-459426578031467f87e167d9a5eb661b    [0.0, 0.0]
8a2af901f154496fbb6036ddd173218b-6b5352d6a195401a90713c628b6d8605    [0.0, 0.0]
9d0c31295c094a9cabd8f714118ace88-93e5a776aaed4da79afb70e1dff36532    [0.0, 0.0]
9d1d675269854f02a228920c94e3e6e9-68475d05a8e24a35b103790b1645c069    [0.0, 0.0]
af10ebce60a54b44bc652acb93fe7585-3bbfaaf3a2954510bc555c0ecc87e18a    [0.0, 0.0]
af154efb5a6e49cb92f84282f3707767-774ab5f2fd304c589299e4f710c630fd    [1.0, 0.0]
b0e8e9e00bc7457c9076e467350695a9-8c5d40afb50645ab82312716ef48fbaa    [0.0, 0.0]
d761dc9f45e14df2a32

In [16]:
df_mm_conversation.groupby(['conversation_pair_id']).agg(list)["useful"]

conversation_pair_id
0603f098136d4d8ab65f6595aedca9ea-254c704903e74ee5b340b3fd3ab7ee61    [0.0, 0.0]
1492d6c6561e4b8286f3ec671f1882fc-c7d7145f38c54446b15489c71c93d665    [0.0, 1.0]
2e0b78151e514f348a779e6d8080c704-12f53562110e4b168191fdb2499b5df4    [1.0, 1.0]
3fb00608c09d4aa9b3d09a9a1c25ad40-d466ed4f141143ad9ac4d1dc126bca1e    [1.0, 1.0]
747e806a957042c19f0708463039e189-c56bb7f6a7584f26873757cdbd95a249    [0.0, 0.0]
842c71afa38c4e5ebbecc439cde2e2df-459426578031467f87e167d9a5eb661b    [1.0, 0.0]
8a2af901f154496fbb6036ddd173218b-6b5352d6a195401a90713c628b6d8605    [0.0, 0.0]
9d0c31295c094a9cabd8f714118ace88-93e5a776aaed4da79afb70e1dff36532    [1.0, 1.0]
9d1d675269854f02a228920c94e3e6e9-68475d05a8e24a35b103790b1645c069    [0.0, 0.0]
af10ebce60a54b44bc652acb93fe7585-3bbfaaf3a2954510bc555c0ecc87e18a    [0.0, 0.0]
af154efb5a6e49cb92f84282f3707767-774ab5f2fd304c589299e4f710c630fd    [1.0, 0.0]
b0e8e9e00bc7457c9076e467350695a9-8c5d40afb50645ab82312716ef48fbaa    [0.0, 0.0]
d761dc9f45e14df2a32

### Réponse

**Hypothèse 1 — Subjectivité inhérente du concept de créativité.**  
La créativité est une notion intrinsèquement subjective. Contrairement à *useful* ("la réponse a-t-elle aidé l'utilisateur ?") qui peut se vérifier factuellement, juger si une réponse est "créative" dépend des attentes esthétiques, du parcours culturel et des goûts personnels de l'annotateur. Deux annotateurs peuvent donc être en total désaccord tout en étant chacun cohérent avec sa propre définition interne de la créativité.

**Hypothèse 2 — Faible prévalence et sensibilité au contexte conversationnel.**  
Le signal *creative* est plus rare car le dataset est un dataset de conversation entre utilisateur et llm et il arrive plus souvent d'avoir des réponse utile que créative. Donc on a un gros déséquilibre et donc un désaccord pèse plus lourd sur la formule du k qui est:
$$\kappa = \frac{p_o - p_e}{1 - p_e}$$
où $p_o$ est la proportion d'accord observé et $p_e$ la proportion d'accord attendu par hasard. Avec un signal rare, même un petit désaccord peut faire chuter $p_o$ de manière significative, tandis que $p_e$ reste relativement élevé, ce qui entraîne un κ négatif. De plus, la créativité peut être perçue différemment selon le contexte de la conversation, rendant l'accord encore plus difficile à atteindre.
